Framework/DataObjects/src/Workspace2D.cpp

```diff
@@ -138,9 +139,15 @@ size_t Workspace2D::size() const { return data.size() * blocksize(); }
 
 /// get the size of each vector
 size_t Workspace2D::blocksize() const {
-  return (!data.empty())
-             ? static_cast<ISpectrum const *>(data[0])->dataY().size()
-             : 0;
+  if (data.empty()) {
+    return 0;
+  } else {
+    size_t numBins = data[0]->size();
+    const auto numHist = this->getNumberHistograms();
+    for (size_t i = 1; i < numHist; ++i)
+      numBins = std::max(numBins, data[i]->size());
```

Comment:
Given that many algorithms that use `blocksize` might be assuming that all histograms have the same size, wouldn't it make more sense to throw an exception upon mismatch? That we we could catch cases where algorithm do not support variable sizes yet. If we simply return the maximum it seems like we could corrupt data easily...?

Reply:
That is what all of the other changes were about. I looked over the various places that `blocksize` was used and changed to more appropriate methods.

Reply:
Ok, if you are certain that you covered all cases. But at the very least I suppose the method should be renamed to `maximumHistogramSizeInWorkspace` or something like that, and the Doxygen be fixed.